internal/clients/kubernetes_client_test.go

```diff
@@ -0,0 +1,241 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/NCCloud/metadata-reflector/internal/common"
+	mockCache "github.com/NCCloud/metadata-reflector/mocks/sigs.k8s.io/controller-runtime/pkg/cache"
+	mockClient "github.com/NCCloud/metadata-reflector/mocks/sigs.k8s.io/controller-runtime/pkg/client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	_ "sigs.k8s.io/controller-runtime/pkg/cache"
+	realClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestNewKubernetesClient(t *testing.T) {
+	config := common.NewConfig()
+	mgr, _ := ctrl.NewManager(&rest.Config{}, ctrl.Options{})
+
+	client := NewKubernetesClient(mgr, config)
+
+	assert.NotNil(t, client)
+}
+
+func TestKubernetesClient_ListDeployments(t *testing.T) {
+	ctx := context.Background()
+	config := common.NewConfig()
+	mockCache := new(mockCache.MockCache)
+	mockClient := new(mockClient.MockClient)
+
+	labelSelector, _ := labels.Parse("matching=true")
+
+	matchingDeployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "matching-deployment",
+			Labels: map[string]string{"matching": "true"},
+		},
+	}
+
+	notMatchingDeployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "not-matching-deployment",
+			Labels: map[string]string{"hello": "world"},
+		},
+	}
+
+	expectedDeploymentList := &appsv1.DeploymentList{
+		Items: []appsv1.Deployment{
+			matchingDeployment,
+		},
+	}
+
+	deploymentList := []appsv1.Deployment{
+		matchingDeployment,
+		notMatchingDeployment,
+	}
+
+	// simulate filtering logic in the mock
+	mockCache.On("List", mock.Anything, mock.Anything, mock.Anything).
+		Return(func(ctx context.Context, list realClient.ObjectList, opts ...realClient.ListOption) error {
+			if depList, ok := list.(*appsv1.DeploymentList); ok {
+				for _, deployment := range deploymentList {
+					if labelSelector.Matches(labels.Set(deployment.Labels)) {
+						depList.Items = append(depList.Items, deployment)
+					}
+				}
+			}
+			return nil
+		})
+
+	client := &kubernetesClient{
+		cacheClient: mockCache,
+		client:      mockClient,
+		config:      config,
+	}
+
+	result, listErr := client.ListDeployments(ctx, labelSelector)
+
+	fmt.Println(len(result.Items), labelSelector.String())
```

Comment:
This can confuse the test result output, did you need this for debug purposes?

Reply:
you are right, thank you!